content/master/guides/composition-functions.md

```diff
@@ -0,0 +1,488 @@
+create slightly better rootless containers. If `xfn` is not granted these
+capabilities it will not support multiple users and groups inside Composition
+Function containers; only UID and GID 0 will exist. Regardless of capabilities
+`xfn` always runs each Composition Function as an unprivileged user. That user
+will appear to be root inside the Composition Function container thanks to 
+[`user_namespaces(7)`].
+
+## Using Composition Functions
+
+To use Composition Functions you must:
+
+1. Find one or more Composition Functions, or write your own.
+2. Write a `Composition` that uses your Composition Functions.
+3. Create an XR that uses your `Composition`.
+
+Your XRs, claims, and providers don't need to be updated or otherwise aware
+of Composition Functions to use them - they need only use a Composition that
+includes one or more entries in its `spec.functions` array.
+
+Composition Functions are designed to be run in a pipeline, so you can 'stack'
+several of them together. Each function is passed the output of the previous
+function as its input. Functions can also be used in conjunction with P&T
+Composition (i.e. a `spec.resources` array). 
+
+In the following example P&T Composition composes an RDS instance. A pipeline of
+(hypothetical) Composition Functions then updates that RDS instance (before it's
+applied to the API server) with a randomly generated password, and composes an
+RDS security group.
+
+```yaml
+apiVersion: apiextensions.crossplane.io/v2alpha1
+kind: Composition
+metadata:
+  name: example
+spec:
+  compositeTypeRef:
+    apiVersion: database.example.org/v1alpha1
+    kind: XPostgreSQLInstance
+  resources:
+    - name: rds-instance
+      base:
+        apiVersion: rds.aws.upbound.io/v1beta1
+        kind: Instance
+        spec:
+          forProvider:
+            dbName: exmaple
+            instanceClass: db.t3.micro
+            region: us-west-2
+            skipFinalSnapshot: true
+            username: exampleuser
+            engine: postgres
+            engineVersion: "12"
+      patches:
+        - fromFieldPath: spec.parameters.storageGB
+          toFieldPath: spec.forProvider.allocatedStorage
+      connectionDetails:
+        - type: FromFieldPath
+          name: username
+          fromFieldPath: spec.forProvider.username
+        - type: FromConnectionSecretKey
+          name: password
+          fromConnectionSecretKey: attribute.password
+  functions:
+  - name: rds-instance-password
+    type: Container
+    container:
+      image: xpkg.upbound.io/provider-aws-xfns/random-rds-password:v0.1.0
+  - name: create-dbsecuritygroup
+    type: Container
+    container:
+      image: xpkg.upbound.io/example-org/create-rds-securitygroup:v0.9.0
+```
+
+
+Use `kubectl explain` to explore the configuration options available when using
+Composition Functions, or take a look at the example Composition below.
+
+{{< expand "Using `kubectl explain` on a Composition" >}}
```

Comment:
Hrm this one doesn't seem to work when rendered. Wonder if the backticks are breaking it. (They also don't render as `code`).

Reply:
Yeah, seems to be the same for the below expand blocks.

Reply:
I haven't tried  backticks in the expand tag. My gut says it probably won't work.  Everything inside the expand should render as normal.